9 create auction page

Mezsoa opened the pull request:
## Done

Finished layout design for the Create Auction page mobile not a finished Desktop design.
I created this PR so that we easily can 
discuss how we want the desktop version to look like.

no functionality is added yet!

## What to check:

take a look at the responsive design/ desktop view-port.

look at the upload design, decide if that's the kind of click/hover behavior we want on either 490x844 & 1920x1080

### components Folder:

  -auctionCreationContainer

Files:

- [ ] AuctionSelectionDropdowns.css//AuctionSelectionDropdowns.jsx
- [ ] FrameBottom.css//FrameBottom.jsx
- [ ] PublishButton.css//PublishButton.jsx


---------------------------------------------------------------------------------------------
### Sub-Folder:

  -checkboxRandomPicture

Files:
- [ ] RandomPic.css//RandomPic.jsx

---------------------------------------------------------------------------------------------
### Sub-Folder:

  -infoFields

Files:
- [ ] InfoFieldsCreateAuction.css//InfoFieldsCreateAuction.jsx
- [ ] InputFields.css//InputFields.jsx
- [ ] UploadIcon.css//UploadIcon.jsx

---------------------------------------------------------------------------------------------


## Extra

REMINDER:
take a look at the desktop version, see if that's something we are ok with or if we want something else.
maybe discuss this with +2 people before deciding.


Upload Icon 

![Screenshot from 2024-05-03 18-00-37](https://github.com/NybergViktor/Celemo_FrontEnd/assets/136836429/2c157940-7207-4540-a64b-dcdd3ae7c4ff)


Closes #9

johnsson84 reviewed (commented):
Looks really  good, some thoughts on desktop size.

![c-auction](https://github.com/NybergViktor/Celemo_FrontEnd/assets/69533976/c6991016-a7d8-40d7-925c-f26999c0b01e)

![c-auction2](https://github.com/NybergViktor/Celemo_FrontEnd/assets/69533976/6031303e-b589-492f-ad66-c44fb4e7b262)

Maybe set a max width and get components together so scrolling is not necessary on desktop? Discuss...

NybergViktor reviewed (changes requested):
# Bra jobbat John
Det syns att du har lagt ner mycket tid på detta.

Håller med Johan om att det kan vara lämpligt att sätta en maxWidth enligt Johans bilder.

Några andra funderingar kring din upload funktion:
     Har du funderat på hur denna funktion ska fungera? 
- Är det tanken att bilden som laddas upp ska fylla upload rutan, isåfall varför ska upload ikonen ändra färg? Blir inte den ersatt av bilden?
- Eller tänker du att den ska hamna under i "text-format"? Isåfall ska man få se om/vad en fil har laddats upp? likt -> 
![Screenshot 2024-05-04 at 20 49 41](https://github.com/NybergViktor/Celemo_FrontEnd/assets/144717637/56ea299a-08de-4e0e-8e5c-234b5601952d)

Oavsett hur du väljer att göra så tror jag vi vill bestämma vilka filtyper som är ok, tex 
 "input> type="file" accept="image/gif,image/jpeg,image/jpg,image/png" > </input" m.m.

Jag vet att du inte börjat med funktion ännu men tänker att design och funktion hänger ihop. 
Angående funktionen har du en riktig utmaning men jag tvivlar inte på att du ror det i hamn! 😃 


Fin PR också, uppskattar tydlighet!


---


Bra Jobbat John!

Mezsoa commented:
## Thanks for the good reviews! 😃 


### Changes made for mobile view

- Instead of a Random Picture checkbox its now a input type="file" so that the user gets prompted to file manager.
- Made it accessible to click for both the icon square and the browser input.
- Added required filtype to upload = image/*, .jpg, .png, .pdf
- Added margin-bottom to input-container so its equal space between everything 

![Screenshot from 2024-05-05 20-02-17](https://github.com/NybergViktor/Celemo_FrontEnd/assets/136836429/2e7da082-0f8f-4edf-af2a-8d449448ffdd)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------

### Changes made for desktop view

- Instead of a Random Picture checkbox its now a input type="file" so that the user gets prompted to file manager.
- Changed the layout design, added margins so its equal everywhere
- Added box-shadow on upload icon

![Screenshot from 2024-05-05 20-18-37](https://github.com/NybergViktor/Celemo_FrontEnd/assets/136836429/bed7b6ea-1a2f-4b14-aea0-b2e77b412207)

NybergViktor reviewed (approved):
Ser bra ut både i mobil och desktop.

Upload funktionen fungerar, man ser vilken fil man valt och man kan ändra både via ikonen och knappen. Man kan inte heller ladda upp js filer och annat. 👍 

Dock får man ingen förhandsvisning hur bilden kommer ser ut men det har vi inte bestämt sedan innan så det spelar egentligen ingen roll.

# Snyggt jobbat John

approved by me

😄
